Docs/src/instancing.md

```diff
@@ -0,0 +1,373 @@
+There will be minimal gains when using instancing on an Entity that repeats very little, or if each instance actually has a different material, or it could run even slower if the Entity never repeats.
+
+If the bottleneck in your game is not CPU (i.e. it's in the GPU) then instancing won't make a noticeable difference.
+
+## Instances per batch {#InstancesPerBatch}
+
+As explained in the previous section, instancing groups all instances into one draw call. 
+However this is half the truth. 
+Instancing actually groups a certain number of instances into a batch. 
+One batch = One draw call.
+
+If the technique is using 80 instances per batch; then rendering 160 instances is going to need 2 draw calls (two batches); 
+if there are 180 instances, 3 draw calls will be needed (3 batches).
+
+What is a good value for instances-per-batch setting? 
+That depends on a lot of factors, which you will have to profile. 
+Normally, increasing the number should improve performance because the system is most likely CPU bottleneck. 
+However, past certain number, certain trade offs begin to show up:
+
+-   Culling is first performed at batch level, then for HW techniques culling is also done at per instance level. 
+    If the batch contains too many instances, its AABB will grow too large; thus the hierarchy culling will always pass and Ogre won't be able skip entire batches.
+-   If the instance per batch is at 10.000 and the application created 10.001 instances; 
+    a lot of RAM & VRAM will be wasted because it's set for 20.000 instances; 
+	HW techniques will spent an excessive amount of CPU time parsing the 9.999 inactive instances; 
+	and SW techniques will saturate the Bus bandwidth sending null matrices for the inactive instances to the GPU.
+
+The actual value will depend a lot on the application and whether all instances are often on screen or frustum culled and whether the total number of instances can be known at production time (i.e. environment props). 
+Normally numbers between 80 and 500 work best, but there have been cases where big values like 5.000 actually improved performance.
+
+# Techniques {#InstancingTechniques}
+
+Ogre supports 4 different instancing techniques. Unfortunately, each of them requires a different vertex shader, since their approaches are different. 
+Also their compatibility and performance varies.
+
+## ShaderBased {#InstancingTechniquesShaderBased}
+
+This is the most compatible technique. 
+It is a Software Instancing technique. 
+World matrices are passed through constant registers, and thus the maximum number of instances per batch is 80; which quickly goes down if the object is skeletally animated. 
+This technique does not play very well with skeletal animation because of that, unless the number of bones is very low (3 or less).
+
+See `material Examples/Instancing/ShaderBased` for an example on how to write the vertex shader. 
```

Comment:
this paragraph is not accurate for Ogre1. The according material would be `Examples/Instancing/RTSS/Robot` - however there are no shaders as they are generated by the RTSS.

Reply:
I understand, but how is someone who wants to use shaders be able to modify their own shaders to use instancing from an example?

I had to download an OGRE 1.8 source to get these examples...

I made a mini guide on which parameters to pass to the vertex shader, so perhaps it could be replaced by that.

Reply:
you could use `ApplicationContextBase::setRTSSWriteShadersToDisk` which will dump the generated shaders and log which pass they belong to.
This is of course not very user-friendly.

However, I think the documentation should focus on the straightforward way and the advanced usage should go in a separate chapter/ tutorial.